### DISCUSSION TITLE
I want to know how long it will take for cookies to expire and whether they need to be retrieved after they expire.

### DISCUSSION THREAD

**Discussion by xingpingcn** [Amm1rr/WebAI-to-API#33] [Q&A] [answered]
I want to know how long it will take for cookies to expire and whether they need to be retrieved after they expire.

**Comment by Amm1rr** [comment, marked as answer]
The cookies expire every 14 days.
To refresh them, just open the chatbot website in a browser.
That's all.

Additionally, the sessions are available to view at [localhost:8000/webai](http://localhost:8000/webai):

[![Image](https://raw.githubusercontent.com/Amm1rr/WebAI-to-API/master/assets/WebAI-Configuration-Thumb.jpg)](https://raw.githubusercontent.com/Amm1rr/WebAI-to-API/master/assets/WebAI-Configuration.jpg)

**Reply by xingpingcn** [reply]
> To refresh them, just open the chatbot website in a browser.

Do you mean that if the session key expires, you only need to open the official chatbot webpage and then you don't need to copy the cookie to the config again?

**Reply by Amm1rr** [reply]
Yes, that's correct.
Actually, copying session keys to the config file is another method to use. The easiest way is to just open the chatbot website with web browsers.

**Reply by xingpingcn** [reply]
After 14 days, I reopened the browser. At this time, I did not need to refresh the cookies manually. Does this mean that the program automatically refreshed them for me? More specifically, I want to know whether the cookies from 14 days ago are the same as those from 14 days later.

**Reply by Amm1rr** [reply]
Yes, everything happens automatically without manual configuration. When you open the website after 14 days, cookies are automatically set and configured, and they are different from the ones 14 days ago.

**Comment by xingpingcn** [comment]
i wonder if there exists a method that can refresh session token without browser or headless browser in order to deploy on FaaS or my vps to access the chat as if i were using the official API
